compiler/env/OMRKnownObjectTable.cpp

```diff
@@ -126,6 +145,26 @@ OMR::KnownObjectTable::getIndexAt(uintptr_t *objectReferenceLocation, bool isArr
    return result;
    }
 
+TR::KnownObjectTable::Index
+OMR::KnownObjectTable::getOrCreateIndexAt(uintptr_t *objectReferenceLocation)
+   {
+#ifdef J9_PROJECT_SPECIFIC
+   TR::VMAccessCriticalSection getIndexCriticalSection(self()->comp());
+#endif
+   uintptr_t objectPointer = *objectReferenceLocation; // Note: object references held as uintptr_t must never be compressed refs
+   Index result = self()->getOrCreateIndex(objectPointer);
+   return result;
+   }
+
+TR::KnownObjectTable::Index
+OMR::KnownObjectTable::getOrCreateIndexAt(uintptr_t *objectReferenceLocation, bool isArrayWithConstantElements)
+   {
+   Index result = self()->getOrCreateIndexAt(objectReferenceLocation);
+   if (isArrayWithConstantElements)
+      self()->addArrayWithConstantElements(result);
+   return result;
+   }
```

Comment:
Does the definition of these functions really need to be in OMR? I think it would make more sense to move these to OpenJ9 and leave the OMR definitions as `TR_UNIMPLEMENTED()`.

Reply:
`getOrCreateIndexAt()` will replace `getIndexAt()` in the next phase. The OpenJ9 implementation is different, which considers `JITServer` mode. Meanwhile, `getIndexAt()` is called in the OMR code which means `getOrCreateIndexAt()` will be called in the OMR code.

Reply:
Hm, I see. So the code is just copy and pasted?

Reply:
Pretty much, except that inside `getOrCreateIndexAt()`, it calls `getOrCreateIndex()` instead of `getIndex()`, since `getIndex()` and `getIndexAt()` will be replaced by the new functions in the next phase. The reason of this change required is to make it clear to the caller of the functions that these methods change the known object table size instead of being a simple getter. I'll update the commit message next to add the reasoning.